tests/fetch_test.ts

```diff
@@ -0,0 +1,161 @@
+import { fetchJSON } from "../src/fetch.ts";
+import { assert, assertEquals, mf } from "./test_deps.ts";
+
+Deno.test("fetchJSON - Returns successful response", async () => {
+  mf.install(); // mock out calls to `fetch`
+
+  mf.mock("GET@/api", () => {
+    return new Response(JSON.stringify({ message: "Success" }), {
+      status: 200,
+      statusText: "OK",
+    });
+  });
+
+  const request = new Request("https://example.com/api");
+
+  const result = await fetchJSON(request);
+  assertEquals(result, { ok: true, data: { message: "Success" } });
+  mf.reset();
+});
+
+Deno.test("fetchJSON - Returns parseError on Bad Request", async () => {
+  mf.install(); // mock out calls to `fetch`
+
+  mf.mock("GET@/api", () => {
+    return new Response(JSON.stringify({ error: "Bad Request" }), {
+      status: 400,
+      statusText: "Bad Request",
+    });
+  });
+
+  const request = new Request("https://example.com/api");
+  // deno-lint-ignore no-explicit-any
+  const result = await fetchJSON(request) as any;
+  assertEquals(result.ok, false);
+  assert(result.message !== undefined);
```

Comment:
Not a specific message we look for here? We just test that result.ok === false and move on?

Reply:
The whole test is a bit weak to be honest..